examples/tensor_base_example.py

```diff
@@ -0,0 +1,117 @@
+"""
+Example showing how to use the no_header mode with a TensorBase CUDA extension
+
+This example creates a CUDA extension that directly includes ATen/core/TensorBase.h 
+instead of torch/extension.h, resulting in faster compilation with no_header=True
+"""
+import torch
+import torch.utils.cpp_extension
+
+# C++ code that directly includes TensorBase.h without using torch/extension.h
+cpp_source = """
+#include <ATen/core/TensorBase.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <pybind11/pybind11.h>
+
+// Forward declaration of the CUDA kernel function
+at::TensorBase tensor_base_add(const at::TensorBase& x, const at::TensorBase& y);
+
+PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
+    m.def("tensor_base_add", &tensor_base_add, "Add two tensors using TensorBase directly");
+}
+"""
+
+# CUDA source with direct TensorBase usage
+cuda_source = """
+#include <ATen/core/TensorBase.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+__global__ void tensor_base_add_kernel(float* __restrict__ x,
+                                    float* __restrict__ y,
+                                    float* __restrict__ out,
+                                    const int size) {
+    const int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    if (idx < size) {
+        out[idx] = x[idx] + y[idx] + 1.0f; // Add 1 to make it distinguishable
+    }
+}
+
+at::TensorBase tensor_base_add(const at::TensorBase& x, const at::TensorBase& y) {
+    // Validate inputs
+    TORCH_CHECK(x.is_cuda(), "x must be a CUDA tensor");
+    TORCH_CHECK(y.is_cuda(), "y must be a CUDA tensor");
+    TORCH_CHECK(x.scalar_type() == at::ScalarType::Float, "x must be a float tensor");
+    TORCH_CHECK(y.scalar_type() == at::ScalarType::Float, "y must be a float tensor");
+    TORCH_CHECK(x.sizes() == y.sizes(), "x and y must have the same shape");
```

Comment:
As you are writing an example, please instill the best practices: either use tensor iterator or check that inputs are contigous
```suggestion
    TORCH_CHECK(x.is_contiguous() && y.is_contigous(), "x and y must be constigous tensors");
```

Reply:
@malfet feedback: Check both sizes are equal and contiguous